src/ConstantProductFactory.sol

```diff
@@ -17,10 +19,67 @@ contract ConstantProductFactory {
      */
     ISettlement public immutable settler;
 
+    /**
+     * @notice Sanity check to make sure that `getTradeableOrderWithSignature`
+     * is being called with parameters that were supposed to be handled by this
+     * factory.
+     */
+    error CanOnlyHandleOrdersForItself();
+    /**
+     * @notice Sanity check to make sure that `getTradeableOrderWithSignature`
+     * is being called with parameters that are currently enabled for trading
+     * on the AMM.
+     *
+     * @param computed the hash of the input parameters
+     * @param amm the hash of the parameters that are allowed to be traded on
+     * the AMM. If the hash is empty, then trading for this order has been
+     * disabled and there is currently no open order available
+     */
+    error ParamsHashDoesNotMatchEnabledOrder(bytes32 computed, bytes32 amm);
+
     /**
      * @param _settler The address of the GPv2Settlement contract.
      */
     constructor(ISettlement _settler) {
         settler = _settler;
     }
+
+    /**
+     * @notice This function exists to let the watchtower off-chain service
+     * automatically create AMM orders and post them on the orderbook. It
+     * outputs an order for the input AMM together with a valid signature.
+     * @dev Some parameters are unused as they refer to features of
+     * ComposableCoW that aren't implemented in this contract. They are still
+     * needed to let the watchtower interact with this contract in the same way
+     * as ComposableCoW.
+     * @param amm owner of the order.
+     * @param params `ConditionalOrderParams` for the order; precisely, the
+     * handler must be this contract, the salt can be any value, and the static
+     * input must be the current trading parameters of the AMM.
+     * @return order discrete order for submitting to CoW Protocol API
+     * @return signature for submitting to CoW Protocol API
+     */
+    function getTradeableOrderWithSignature(
+        ConstantProduct amm,
+        IConditionalOrder.ConditionalOrderParams calldata params,
+        bytes calldata, // offchainInput
+        bytes32[] calldata // proof
+    ) external view returns (GPv2Order.Data memory order, bytes memory signature) {
+        if (address(params.handler) != address(this)) {
+            revert CanOnlyHandleOrdersForItself();
```

Comment:
As this will be polled by the watch-tower, and this is essentially an order that's not valid, it should return `OrderNotValid(string)` as the revert message to make sure the watch-tower drops it with the applicable message.